lib/_stream_writable.js

```diff
@@ -465,9 +469,18 @@ function onwrite(stream, er) {
   state.length -= state.writelen;
   state.writelen = 0;
 
-  if (er)
-    onwriteError(stream, state, sync, er, cb);
-  else {
+  if (er) {
+    state.errored = true;
```

Comment:
We cannot use `errorEmitted` since it must be set **after** `cb` and `cb` **must** be invoked asynchronously.

Reply:
We cannot use `destroyed` since it will not be set on `!autoDestroy`,